### PR TITLE
Add feature for usr.sbin/newsyslog.c: accept human unit suffix for size fileds in configuration file.

### PR DESCRIPTION
**Pull request by Rin0913**
In previous versions of newsyslog, the size had to be configured in KB, which is too small for modern computers.

With this update, we can use three types of units: K, M, and G. For example,
```
# logfilename          [owner:group]    mode count size when  flags [/pid_file] [sig_num]
/var/log/all.log                        600  7     *    @T00  J
/var/log/auth.log                       600  7     1000 @0101T JC
/var/log/console.log                    600  5     100M *     J
/var/log/cron                           600  3     1G   *     JC
/var/log/daemon.log                     644  5     1000K @0101T JC
```

### CODE REVIEW

**Comment by lwhsu** [comment]
I think we can use `expand_number(3)` from libutil?

**Comment by lwhsu** [comment]
And we may also need to update manual page to indicate that we can use human readable format.

**Comment by Rin0913** [comment]
The manual will look like this:

image

**Comment by concussious** [comment]
Does this also need a change to newsyslog.8?

**Comment by Rin0913** [comment]
@concussious I don't think so due to its unchanged command line argument. The only thing changed is the configuration and how it parse.

File: usr.sbin/newsyslog/newsyslog.c
Comment (jlduran) [inline usr.sbin/newsyslog/newsyslog.c:1345]: What happens when `expand_number` fails?
Maybe something like:
```c
if (expand_number(q, &trsize) != 0) {
	warnx("Invalid value of '%s' for 'size' in line:\n%s",
	    q, errline);
	working->trsize = -1;
}
```

File: usr.sbin/newsyslog/newsyslog.c
Comment (jlduran) [inline usr.sbin/newsyslog/newsyslog.c:1343]: Gratuitous white space at the end.

File: usr.sbin/newsyslog/newsyslog.conf.5
Comment (jlduran) [inline usr.sbin/newsyslog/newsyslog.conf.5:120]: I would prefer something along the lines:
```mdoc
in kilobytes, or by using any suffix (e.g., k, M, G) supported by
.Xr expand_number 3 ,
the log file will be trimmed as described above.
```
And a cross reference to this manual in the `SEE ALSO` section.

**Review by jlduran — CHANGES_REQUESTED** [review]
Great addition!
Just a few minor suggestions. Thank you!

File: usr.sbin/newsyslog/newsyslog.conf.5
Comment (Rin0913) [reply usr.sbin/newsyslog/newsyslog.conf.5:120]: image
image

Comment (jlduran) [reply usr.sbin/newsyslog/newsyslog.conf.5]: Sorry to insist, but `kilobyte` should be written as is.
From https://en.wikipedia.org/wiki/Kilobyte:
> The internationally recommended unit symbol for the kilobyte is **kB**.[^1]

[^1]: https://en.wikipedia.org/wiki/Kilobyte#cite_note-IEC80000-1

File: usr.sbin/newsyslog/newsyslog.conf.5
Comment (jlduran) [inline usr.sbin/newsyslog/newsyslog.conf.5:120]: Sorry to insist, but `kilobyte` should be written as is.
From https://en.wikipedia.org/wiki/Kilobyte:
> The internationally recommended unit symbol for the kilobyte is **kB**.[^1]

Also:
- `mandoc -T lint` will request that `expand_number(3)` is placed before `syslog(3)`, to keep the list sorted alphabetically by manual.
- [`igor(1)`](https://man.freebsd.org/cgi/man.cgi?igor(1)), will recommend to add a comma after _exempli gratia_.

The manual has other minor imperfections but are not related to this pull request. At any rate, a bump in the date should be required.

[^1]: https://en.wikipedia.org/wiki/Kilobyte#cite_note-IEC80000-1

**Review by jlduran — CHANGES_REQUESTED** [review]
This comment got hidden bu the resolved conversation.

File: usr.sbin/newsyslog/newsyslog.conf.5
Comment (Rin0913) [reply usr.sbin/newsyslog/newsyslog.conf.5]: Sorry for my bad english.
Do you mean that I should write 'kilobyte' instead of 'kilobytes'?

File: usr.sbin/newsyslog/newsyslog.conf.5
Comment (Rin0913) [reply usr.sbin/newsyslog/newsyslog.conf.5]: Sorry for my poor English,
I would like to make sure you want me to use `kilobyte` instead of `kilobytes`?

Comment (Rin0913) [reply usr.sbin/newsyslog/newsyslog.conf.5]: Sorry for my careless behavior,

image

I think it is what you are trying to say?

Comment (jlduran) [reply usr.sbin/newsyslog/newsyslog.conf.5]: > [Edited]
> 
> Sorry for my careless behavior,
> 
> image
> 
> I think it is what you are trying to say?

Yes, the letter “k” in kilobyte, should be lowercased (as it currently is).

File: usr.sbin/newsyslog/newsyslog.conf.5
Comment (jlduran) [reply usr.sbin/newsyslog/newsyslog.conf.5]: > Sorry for my bad english. Do you mean that I should write 'kilobyte' instead of 'kilobytes'?

With the letter "k" (lowercase).

File: usr.sbin/newsyslog/newsyslog.conf.5
Comment (jlduran) [inline usr.sbin/newsyslog/newsyslog.conf.5:120]: The word should be kilobytes (plural and with a lowercase "k"), the same for the prefix (not suffix).

**Review by jlduran — CHANGES_REQUESTED** [review]
Once the documentation change is adjusted, I have no further suggestions, and can be considered a pre-approved.
We should wait for @lwhsu or some other CODEOWNER to actually review it and (hopefully) merge it.
Bear in mind that most developers are busy with 14.2 about to be released, so expect some delays.
Thank you for your contribution

File: usr.sbin/newsyslog/newsyslog.conf.5
Comment (Rin0913) [reply usr.sbin/newsyslog/newsyslog.conf.5]: Sorry for that, I forgot to push.

**Review by jlduran — APPROVED** [review]

**Comment by lwhsu** [comment]
Merged as 1f78bbb5c415efefa782c1be7f9a2e4a5008903b
@jlduran sorry, forgot to include you as `Reviewed-by` in the commit message :(

**Comment by jlduran** [comment]
> Merged as 1f78bbb5c415efefa782c1be7f9a2e4a5008903b @jlduran sorry, forgot to include you as `Reviewed-by` in the commit message :(

No problem at all! Thank you for committing.

**Comment by jlduran** [comment]
@lwhsu somehow the manual page change landed with a duplicate word (`suffix suffixes`).

**Pull request closed without merging**
